include/openthread/thread.h

```diff
@@ -466,6 +466,38 @@ otError otThreadSetMeshLocalPrefix(otInstance *aInstance, const otMeshLocalPrefi
  */
 const otIp6Address *otThreadGetLinkLocalIp6Address(otInstance *aInstance);
 
+/**
+ * This function returns the Thread Link-Local All Thread Nodes multicast address.
+ *
+ * The address is a link-local Unicast Prefix-Based Multcast Address [RFC 3306], with:
+ *   - flgs set to 3 (P = 1 and T = 1)
+ *   - scop set to 2
+ *   - plen set to the Mesh Local Prefix
+ *   - group ID set to 1
+ *
+ * @param[in]  aInstance A pointer to an OpenThread instance.
+ *
+ * @returns A pointer to Thread Link-Local All Thread Nodes multicast address.
+ *
+ */
+const otIp6Address *otThreadGetLinkLocalAllThreadMulticastIp6Address(otInstance *aInstance);
```

Comment:
```suggestion
const otIp6Address *otThreadGetLinkLocalAllNodeMulticastIp6Address(otInstance *aInstance);
```
How about "all node" vs "all thread"?

Reply:
I think 'All Nodes' might be confusing because it may refer to 'ff02::1`. Perhaps we can use `AllThreadNodes` here.

Reply:
I'm OK with `AllThreadNodes`.

And to avoid being too long, I think `Ip6` can be removed as it's implied in the return type.

Reply:
SGTM